### ISSUE TITLE
Verify the time range bounds for InfluxDB query use-cases

### ISSUE THREAD

**Issue by e-dard** [influxdata/influxdb_iox#2734]
Various bits of IOx assume that time ranges on the gRPC API are in the domain `[start, stop)`. For a long time this was the case in InfluxDB. However, I think with Flux that this may no longer be the case. If so then we will need to make some changes to IOx.

I will verify the current state of play with Flux and InfluxQL and update IOx accordingly.

**Comment by e-dard** [comment]
Here is the deal.

TLDR:

 - The InfluxDB Cloud Flux Query Engine still uses the historic time range domain of `[start, stop)`. This is what I expected and the assumption we built into IOx.
 - The InfluxDB Cloud [delete API](https://docs.influxdata.com/influxdb/cloud/write-data/delete-data/) uses a *different* domain on its `start` and `stop` parameters, specifically `[start, stop]`. This is what I suggested to @NGA-TRAN as I was guiding her through the Delete API initiative on IOx.

So it looks like we have implemented everything correctly but it's unfortunate that the InfluxDB Cloud API time range differs in domain from the Flux Query API domain. I will bring this up with product.

**Note** InfluxQL is not relevant to this discussion because you have to explicitly specify the time range predicate in the query (e.g., `time >= 100`) which clearly shows the expected comparator.

## Details

I wrote the following points into a bucket:

```
cpu temp=100.0 1000000000 // 1970-01-01T00:00:01Z
cpu temp=100.0 2000000000 // 1970-01-01T00:00:02Z
cpu temp=100.0 3000000000 // 1970-01-01T00:00:03Z
cpu temp=100.0 4000000000 // 1970-01-01T00:00:04Z
cpu temp=100.0 5000000000 // 1970-01-01T00:00:05Z
```

### Verify Query time range domain

I queried across all the data using Flux. The documentation for Flux's `range` function is [here](https://docs.influxdata.com/flux/v0.x/stdlib/universe/range/). When providing integers to the `range` function they're treated as seconds since the epoch.

```
from(bucket: "test_time")
  |> range(start: 0, stop: 10) // [0, 10) seconds past the epoch
  |> filter(fn: (r) => r["_measurement"] == "cpu")
  |> drop(columns: ["_start", "_stop"])
```

Gives us all five points.
<img width="1411" alt="Screenshot 2021-10-06 at 12 00 07" src="https://user-images.githubusercontent.com/501993/136192391-7e05e8f5-d517-4b3e-b767-ed0aa2193c33.png">

Next I queried with the `start` time being **equal** to the first point in the set. The domain on `start` is inclusive `[start,` so we still get back all the points

<img width="1368" alt="Screenshot 2021-10-06 at 12 00 20" src="https://user-images.githubusercontent.com/501993/136192561-0ad41f1f-2f1b-4129-b406-dce1d2be0f60.png">

Next I changed the `stop` value to be equal to the last point in the set. The domain on `stop` is `exclusive `[start, stop)` so the final point in the set should be filtered out and we only get back four points.

<img width="1368" alt="Screenshot 2021-10-06 at 12 01 04" src="https://user-images.githubusercontent.com/501993/136192739-4104eb82-7896-4e0c-83d3-353347f40d1d.png">

### Verify Delete API time range domain 

Finally I verified the time range bounds on the delete API. The documentation is [here](https://docs.influxdata.com/influxdb/cloud/write-data/delete-data/).

I submitted the following API call:

```
curl --request POST https://us-west-2-1.aws.cloud2.influxdata.com/api/v2/delete/\?org_id\=abc123\&bucket\=test_time \
  --header 'Authorization: Token myToken' \
  --header 'Content-Type: application/json' \
  --data '{
    "start": "1970-01-01T00:00:01Z",
    "stop": "1970-01-01T00:00:04Z", 
    "predicate": "_measurement=\"cpu\""
  }'
```

I think queried all of the data again. I found that only one point remains. The point with the timestamp equal to the `stop` values on the API call has been deleted. 

<img width="1358" alt="Screenshot 2021-10-06 at 12 04 57" src="https://user-images.githubusercontent.com/501993/136193489-51864434-4027-4278-8df7-34ffc65ddc82.png">

Therefore, the domain for the time range on delete queries differs from query, and is `[start, stop]`.

**Comment by NGA-TRAN** [comment]
Oh no. It sounds like we have to keep 2 two different time ranges here
